packages/mgt-components/src/components/mgt-file/mgt-file.ts

```diff
@@ -0,0 +1,657 @@
+      fileIconSrc = this.fileIcon;
+    } else {
+      // get file type extension from file name
+      const re = /(?:\.([^.]+))?$/;
+      const fileType = re.exec(this.driveItem.name)[1] || 'folder';
+      fileIconSrc = getFileTypeIconUriByExtension(fileType, 48, 'svg');
+    }
+
+    return html`
+      <div class="item__file-type-icon">
+        <img src=${fileIconSrc} />
+      </div>
+    `;
+  }
+
+  /**
+   * Render the file details
+   *
+   * @protected
+   * @param {MicrosoftGraph.DriveItem} [driveItem]
+   * @memberof MgtFile
+   */
+  protected renderDetails(driveItem: MicrosoftGraph.DriveItem): TemplateResult {
+    if (!driveItem || this.view === ViewType.image) {
+      return html``;
+    }
+
+    const details: TemplateResult[] = [];
+
+    if (this.view > ViewType.image) {
+      const text = this.getTextFromProperty(driveItem, this.line1Property);
+      if (text) {
+        details.push(html`
+          <div class="line1" aria-label="${text}">${text}</div>
+        `);
+      }
+    }
+
+    if (this.view > ViewType.oneline) {
+      const text = this.getTextFromProperty(driveItem, this.line2Property);
+      if (text) {
+        details.push(html`
+          <div class="line2" aria-label="${text}">${text}</div>
+        `);
+      }
+    }
+
+    if (this.view > ViewType.twolines) {
+      const text = this.getTextFromProperty(driveItem, this.line3Property);
+      if (text) {
+        details.push(html`
+          <div class="line3" aria-label="${text}">${text}</div>
+        `);
+      }
+    }
+
+    return html`
+      <div class="item__details">
+        ${details}
+      </div>
+    `;
+  }
+
+  /**
+   * load state into the component.
+   *
+   * @protected
+   * @returns
+   * @memberof MgtFile
+   */
+  protected async loadState() {
+    const provider = Providers.globalProvider;
+    if (!provider || provider.state === ProviderState.Loading) {
+      return;
+    }
+
+    if (provider.state === ProviderState.SignedOut) {
+      this.driveItem = null;
+      return;
+    }
+
+    const graph = provider.graph.forComponent(this);
+    let driveItem;
+
+    // return null when a combination of provided properties are required
+    if (
+      (this.driveId && (!this.itemId && !this.itemPath)) ||
+      (this.siteId && (!this.itemId && !this.itemPath)) ||
+      (this.groupId && (!this.itemId && !this.itemPath)) ||
+      (this.listId && (!this.siteId && !this.itemId)) ||
+      (this.insightType && !this.insightId) ||
+      (this.userId && (!this.insightType && !this.insightId)) ||
+      (this.userId && (!this.itemId && !this.itemPath))
+    ) {
+      driveItem = null;
+    }
+
+    if (this.fileDetails) {
+      driveItem = this.fileDetails;
+    }
+
+    // evaluate to true when only item-id or item-path is provided
+    const getFromMyDrive = !this.driveId && !this.siteId && !this.groupId && !this.listId && !this.userId;
+
+    // todo: error handle can't find file
+
+    if (this.fileQuery) {
```

Comment:
Should these `if` statements be an `else if` to avoid going down this path if the `fileDetails` are provided or if the required propoerties are not provided? I'd also move the check for `fileDetails` to be even above the check for the provider so if the `fileDetails` are already provided, no need to go to the graph at all.